inc/Smartling/Jobs/JobAbstract.php

```
@@ -210,87 +137,46 @@ public function dropLockFlag()
-                    $this->getWorkerTTL(),
+                    $this->workerTTL,
                 ]
             )
         );
 
-        $this->tryRunJob($source);
+        $this->tryRunJob();
     }
 
-    /**
-     * @param string $source
-     */
-    private function tryRunJob($source)
+    private function tryRunJob(): void
     {
-        $this->transactionManager->setAutocommit(0);
-        $allowedToRun = false;
-        $exception = null;
-        $message = null;
         try {
-            $this->transactionManager->transactionStart();
-            $result = $this->transactionManager->executeSelectForUpdate($this->buildSelectQuery());
-            if (null !== $result) {
-                $currentTS = time();
-                $flagTS = (0 < count($result)) ? (int)$result[0]['ts'] : 0;
-
-                $this->getLogger()->debug(
-                    vsprintf(
-                        'Checking flag \'%s\' for cron job \'%s\'. Stored value=\'%s\'',
-                        [
-                            $this->getCronFlagName(),
-                            $this->getJobHookName(),
-                            $flagTS,
-                        ]
-                    )
-                );
-                $allowedToRun = $currentTS > $flagTS;
-                if ($allowedToRun) {
-                    $this->placeLockFlag();
-                } else {
-                    $message = vsprintf(
-                        'Cron \'%s\' is not allowed to run. TTL=%s has not expired yet. Expecting TTL expiration in %s seconds.',
-                        [
-                            $this->getJobHookName(),
-                            $this->getWorkerTTL(),
-                            $flagTS - time(),
-                        ]
-                    );
-                    $this->getLogger()->debug($message);
-                }
-            }
+            $this->placeLockFlag();
+        } catch (SmartlingApiException $e) {
```

I think we must catch all exceptions here because it's one of plugin "entry point". So ideal structure can be
```php
private function tryRunJob(): void
{
  try {
    $this->placeLockFlag();
    try {
      $this->run();
    // } catch (...) {...} You can add it only if you would like to handle specific exception of add more data into exception
    } finally {
      $this->dropLockFlag();
    }
  } catch (\Exception $exception) {
    // We log ALL exceptions here
  }
}
```

---

Done